operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSourceRegistry.java

```diff
@@ -16,11 +17,12 @@
    *         registered.
    * @throws OperatorException if an error occurred during the registration process
    */
-  void registerEventSource(EventSource eventSource)
+  void registerEventSource(EventSource<T> eventSource)
       throws IllegalStateException, OperatorException;
 
-  Set<EventSource> getRegisteredEventSources();
+  Set<EventSource<T>> getRegisteredEventSources();
 
   ControllerResourceEventSource<T> getControllerResourceEventSource();
 
+  EventHandler getEventHandler();
```

Comment:
Why should be the registry aware of the handler?

Reply:
All `EventSources` registered with the same `EventRegistry` should use the same `EventHandler` so that's how they get which `EventHandler` to use. This makes the link between the registry, sources and event handler slightly more explicit. I do agree that this is kind of not strictly related to the pure registry aspects of the interface. The problem is that the abstraction is not really appropriate to reflect the reality of the implementation.